### PR TITLE
feat(server): implement json.arrpop command (#104)

### PR DESCRIPTION
**Pull request by iko1**
Signed-off-by: iko1 <me@remotecpp.dev>

### CODE REVIEW

File: src/server/json_family.cc [hunk 1]
```diff
@@ -588,8 +589,101 @@ OpResult<long> OpClear(const OpArgs& op_args, string_view key, string_view path)
...
+    }
+
+    if (removal_index >= val.size()) {
+      removal_index %= val.size();  // rounded to the array boundaries.
+    }
+
+    size_t i = 0;
+    for (auto it = val.array_range().begin(); it != val.array_range().end(); it++) {
+      if (i == removal_index) {
+        ostringstream os;
```
Comment (romange) [inline src/server/json_family.cc:622]: I wonder - is it the only way to convert elements into strings? i.e. only with ostringstream ?

File: src/server/json_family.cc [hunk 2]
```diff
@@ -588,8 +589,101 @@ OpResult<long> OpClear(const OpArgs& op_args, string_view key, string_view path)
...
+    } else {
+      removal_index = index;
+    }
+
+    if (removal_index >= val.size()) {
+      removal_index %= val.size();  // rounded to the array boundaries.
+    }
+
+    size_t i = 0;
+    for (auto it = val.array_range().begin(); it != val.array_range().end(); it++) {
```
Comment (romange) [inline src/server/json_family.cc:620]: there is no way to just access an element at index `i`? you must do the iteration yourself?

**Review by romange — COMMENTED** [review]
lgtm, some minor comments

File: src/server/json_family.cc (hunk 2 above)
Comment (iko1) [reply src/server/json_family.cc:620]: This was the first thing that I've checked and unfortunately the library does not support access to individual members within an array.

File: src/server/json_family.cc (hunk 1 above)
Comment (iko1) [reply src/server/json_family.cc:622]: It's possible to dump JSON object to an object that implements `std::basic_ostream<char_type>`

File: src/server/json_family.cc (hunk 2 above)
Comment (romange) [reply src/server/json_family.cc:620]: well, it that case, maybe `auto it = std::next(val.array_range().begin(), removal_index);'` seems like a better choice.

Comment (iko1) [reply src/server/json_family.cc]: Fixed

**Review by romange — APPROVED** [review]

**Pull request merged**